src/gf.c

```diff
@@ -126,7 +126,7 @@ const struct jl_typemap_info method_defs = {
     0, &jl_method_type
 };
 const struct jl_typemap_info lambda_cache = {
-    0, &jl_method_instance_type
+    1, &jl_method_instance_type
```

Comment:
Sorry, when reviewing earlier I mistook which cache this was and thought it was the tfunc / specializations cache. The lambda (method table) cache requires sorting.